rgw/v2/tests/s3_swift/reusable.py

```diff
@@ -532,22 +535,25 @@ def put_get_bucket_lifecycle_test(bucket, rgw_conn, rgw_conn2, life_cycle_rule,
     else:
         raise TestExecError("bucket life cycle retrieved")
     objs_total = (config.test_ops["version_count"]) * (config.objects_count)
+    time_diff = math.ceil(time.time() - upload_start_time)
+    time_limit = upload_start_time + (config.rgw_lc_debug_interval * 20)
     for rule in config.lifecycle_conf:
         if rule.get("Expiration", {}).get("Date", False):
             # todo: need to get the interval value from yaml file
             log.info("wait for 60 seconds")
             time.sleep(60)
         else:
-            for time_interval in range(19):
+            while time.time() < time_limit:
                 bucket_stats_op = utils.exec_shell_cmd(
                     "radosgw-admin bucket stats --bucket=%s" % bucket.name
                 )
                 json_doc1 = json.loads(bucket_stats_op)
                 obj_pre_lc = json_doc1["usage"]["rgw.main"]["num_objects"]
                 if obj_pre_lc == objs_total:
-                    time.sleep(30)
+                    time.sleep(config.rgw_lc_debug_interval)
                 else:
                     raise TestExecError("Objects expired before the expected days")
+            time.sleep(time_diff)
             time.sleep(60)
```

Comment:
Can we make it single line `time.sleep(60 + time_diff)`

Reply:
done uday!